### ISSUE TITLE
Support for metadata

### ISSUE THREAD

**Issue by rubdos** [marshmallow-code/marshmallow-jsonapi#28]
As in the example on the [documentation](http://jsonapi.org/format/#document-meta), I'd like to add copyright and license information to my JSONAPI objects. It would be nice if the meta top-level objects would be supported.

``` javascript
{
  "meta": {
    "copyright": "Copyright 2015 Example Corp.",
    "authors": [
      "Yehuda Katz",
      "Steve Klabnik",
      "Dan Gebhardt",
      "Tyler Kellen"
    ]
  },
  "data": {
    // ...
  }
}
```

I'm interested in doing this, any remarks before I begin?

**Comment by sloria** [comment]
Thanks for the PR.

How is this different from using a `Dict` field that is `dump_only`?

**Comment by allistercsmith** [comment]
I am looking at adding something to allow for arbitrary top-level meta objects, not just fields set on the schema and/or model.

My main driver is for something like extra pagination information. There's no rule in the spec about where to place things such as offsets, limits and counts, so it makes sense to put these in the meta object. There's actually an example of this in the extensions section [here](http://jsonapi.org/extensions/), under _Profiles_. I imagine it would be useful for a lot of other cases too.

I'm thinking that I could perhaps override the dump method on the schema, and add an optional dictionary parameter to build a meta object from. What do you guys think?

**Comment by rubdos** [comment]
> I am looking at adding something to allow for arbitrary top-level meta objects, not just fields set on the schema and/or model.

I guess that's the better implementation indeed, but keep in mind that you could be making it overly complicated. I don't really know if you _would_ though, as I didn't dive very deep into the code.

IIRC, this implementation isn't feature complete yet, so any patch would be great. I found this library very useful as a base for an API.

> I'm thinking that I could perhaps override the dump method on the schema, and add an optional dictionary parameter to build a meta object from. What do you guys think?

I have actually no idea, it's been too long (only a month, I'm confused) since I wrote and used this. I'm currently on another project (something totally different), but feel free to let me know about this, as I probably will return on using `marshmallow-jsonapi`.

**Comment by sloria** [comment]
Closed by #56
